internal/manager/init.go

```diff
@@ -38,13 +38,10 @@ func Initialize(cfg *config.Config, l *log.Logger) (*Manager, error) {
 	var db *sqlite.Database
 
 	{
-		var dbType = sqlite.DatabaseType(strings.ToUpper(cfg.GetDatabaseType()))
+		var dbType = sqlite.SqliteBackend
 
-		switch dbType {
-		case sqlite.SqliteBackend, sqlite.PostgresBackend:
-			// Valid case
-		default:
-			dbType = sqlite.SqliteBackend
+		if strings.HasPrefix(strings.ToUpper(cfg.GetDatabaseConnectionString()), string(sqlite.PostgresBackend)) {
```

Comment:
You may want to check for for prefix + `:` or this would match a file name of "postgres.sqlite" (unlikely, but possible)

```suggestion
		if strings.HasPrefix(strings.ToUpper(cfg.GetDatabaseConnectionString()), string(sqlite.PostgresBackend) + ":") {
```

Reply:
GetDatabasePath is its own seperate function that returns the config entry "database" that has the database path (for sqlite). 
Where GetDatabaseConnectionString is "database_string" which has a connection string for pgsql.

Reply:
maybe we could simplify this even further

- If the `database` config option begins with `postgres:`, then we use postgres and parse it as URL
- If it begins with `sqlite:` we use SQLite (and the path to file after, e.g. `sqlite:/path/to/file.db` or `sqlite:file.db`)
- If it doesn't have any prefix, then we assume it's a SQLite DB path (e.g. `file.db`) - this is for backwards-compatibility

Then we need a single config option only

Reply:
https://github.com/NodudeWasTaken/stash/pull/1